### PR TITLE
feat: ✨ Added show time on live timeline

### PR DESCRIPTION
**Pull request by DevyankShaw**
# Description

This PR is adding optional functionality that allows users to show time on the live timeline indicator in different views

### Earlier

image

### Now

<details><summary>On activating _showTime_ only</summary>
<p>

- **DayView**

```
DayView<Event>(
      key: state,
      width: width,
      verticalLineOffset: 0,
      timeLineWidth: 65,
      showLiveTimeLineInAllDays: true,
      liveTimeIndicatorSettings: LiveTimeIndicatorSettings(
        color: Colors.redAccent,
        showTime: true,
      ),
    )
```

- **WeekView**

```
WeekView<Event>(
      key: state,
      width: width,
      showLiveTimeLineInAllDays: true,
      timeLineWidth: 65,
      liveTimeIndicatorSettings: LiveTimeIndicatorSettings(
        color: Colors.redAccent,
        showTime: true,
      ),
    )
```

</p>
</details>

image

<details><summary>On activating _showTime_ and _showTimeBackgroundView_</summary>
<p>

- **DayView**
```
DayView<Event>(
      key: state,
      width: width,
      verticalLineOffset: 0,
      timeLineWidth: 65,
      showLiveTimeLineInAllDays: true,
      liveTimeIndicatorSettings: LiveTimeIndicatorSettings(
        color: Colors.redAccent,
        showTime: true,
        showTimeBackgroundView: true,
      ),
    )
```

- **WeekView**

```
WeekView<Event>(
      key: state,
      width: width,
      showLiveTimeLineInAllDays: true,
      timeLineWidth: 65,
      liveTimeIndicatorSettings: LiveTimeIndicatorSettings(
        color: Colors.redAccent,
        showTime: true,
        showTimeBackgroundView: true,
      ),
    )
```
</p>
</details>

image

<details><summary>On activating _showTime_,  _showTimeBackgroundView_ and deactivating _showBullet_</summary>
<p>

- **DayView**
```
DayView<Event>(
      key: state,
      width: width,
      verticalLineOffset: 0,
      timeLineWidth: 65,
      showLiveTimeLineInAllDays: true,
      liveTimeIndicatorSettings: LiveTimeIndicatorSettings(
        color: Colors.redAccent,
        showBullet: false,
        showTime: true,
        showTimeBackgroundView: true,
      ),
    )
```

- **WeekView**
```
WeekView<Event>(
      key: state,
      width: width,
      showLiveTimeLineInAllDays: true,
      timeLineWidth: 65,
      liveTimeIndicatorSettings: LiveTimeIndicatorSettings(
        color: Colors.redAccent,
        showBullet: false,
        showTime: true,
        showTimeBackgroundView: true,
      ),
    )
```
</p>
</details>

image

<details><summary>On activating _showTime_ and deactivating _showBullet_</summary>
<p>

- **DayView**
```
DayView<Event>(
      key: state,
      width: width,
      verticalLineOffset: 0,
      timeLineWidth: 65,
      showLiveTimeLineInAllDays: true,
      liveTimeIndicatorSettings: LiveTimeIndicatorSettings(
        color: Colors.redAccent,
        showBullet: false,
        showTime: true,
      ),
    )
```

- **WeekView**
```
WeekView<Event>(
      key: state,
      width: width,
      showLiveTimeLineInAllDays: true,
      timeLineWidth: 65,
      liveTimeIndicatorSettings: LiveTimeIndicatorSettings(
        color: Colors.redAccent,
        showBullet: false,
        showTime: true,
      ),
    )
```
</p>
</details>

image
 
### Motivation
The designs were approved by the client which had the show time on timeline behavior. I have already implemented the entire day and week view functionality using this package only and don't want to switch to another package just for this behavior. By the way, this is also the core part of the calendar view design as per the latest trends. So thought of creating this PR would help others to choose this package without any hesitation.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have followed the [Contributor Guide] when preparing my PR.
- [ ] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues
#218 

[Contributor Guide]: https://github.com/SimformSolutionsPvtLtd/flutter_calendar_view/blob/master/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

**Comment by DhavalRKansara** [comment]
Hello @DevyankShaw Thanks for you contribution to the package. We would love to test this change and merge it if all goes well in our next release but to achieve the same together could you please resolve the merge conflict in this PR with respects to latest changes on the master?

**Comment by DevyankShaw** [comment]
Hey @DhavalRKansara will try to resolve this merge conflict by this weekend and let you know once done

**Comment by DhavalRKansara** [comment]
> Hey @DhavalRKansara will try to resolve this merge conflict by this weekend and let you know once done

Pefect @DevyankShaw, Thanks for reverting back actively.

**Comment by DevyankShaw** [comment]
@DhavalRKansara merge conflict is now resolved with respect to the master. 

Request to review and let me know if you require any further inputs from my side

File: lib/src/components/_internal_components.dart [hunk 1]
```diff
@@ -127,33 +142,89 @@ class TimeLine extends StatelessWidget {
...
+    super.dispose();
+  }
+
+  /// Creates an recursive call that runs every 1 seconds.
+  /// This will rebuild TimeLine every second. This will allow us
+  /// to show/hide time line when there is overlap with
+  /// live time line indicator in Week and Day view.
+  void setTimer() {
+    if (mounted) {
+      setState(() {
```
Comment (PRBaraiya) [inline lib/src/components/_internal_components.dart:183]: Can we improve this to set the state only if the hour and minutes are actually changed?

Comment (PRBaraiya) [reply lib/src/components/_internal_components.dart:183]: Also, use `Timer.periodic` if possible.

**Review by PRBaraiya — CHANGES_REQUESTED** [review]
Please check the comments and rebase the PR on the latest changes.

File: lib/src/components/_internal_components.dart (hunk 1 above)
Comment (DevyankShaw) [reply lib/src/components/_internal_components.dart:183]: @PRBaraiya Thanks for your feedback. Will try to incorporate those and update once done.

**Comment by ParthBaraiya** [comment]
Screenshot 2023-12-13 at 3 53 57 PM
There are errors in the code.

**Comment by ParthBaraiya** [comment]
Please make sure that all the checks are passed.

**Comment by DevyankShaw** [comment]
It's currently WIP. Will update once done.

**Comment by DevyankShaw** [comment]
@ParthBaraiya Rebasing is done and all checks are passed. I hope it is now ready to merge. Let me know if you require anything.

**Comment by DevyankShaw** [comment]
@ParthBaraiya Can we close this feature at the earliest?

**Review by ParthBaraiya — APPROVED** [review]
LGTM!

**Pull request merged**